src/server/json_family.cc

```diff
@@ -0,0 +1,114 @@
+// Copyright 2022, Roman Gershman.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "server/json_family.h"
+
+extern "C" {
+#include "redis/object.h"
+}
+
+#include "base/logging.h"
+#include "server/command_registry.h"
+#include "server/error.h"
+#include "server/tiered_storage.h"
+#include "server/transaction.h"
+
+#include <jsoncons/json.hpp>
+#include <jsoncons_ext/jsonpath/jsonpath.hpp>
+
+namespace dfly {
+
+using namespace std;
+using namespace jsoncons;
+
+using CI = CommandId;
+
+string GetString(EngineShard* shard, const PrimeValue& pv) {
+  string res;
+  if (pv.IsExternal()) {
+    auto* tiered = shard->tiered_storage();
+    auto [offset, size] = pv.GetExternalPtr();
+    res.resize(size);
+
+    error_code ec = tiered->Read(offset, size, res.data());
+    CHECK(!ec) << "TBD: " << ec;
+  } else {
+    pv.GetString(&res);
+  }
+
+  return res;
+}
+
+OpResult<string> JsonFamily::OpGet(const OpArgs& op_args, string_view key, string_view path) {
+  error_code ec;
+  jsonpath::jsonpath_expression expr = jsonpath::make_expression<json>(path, ec);
```

Comment:
move `jsonpath_expression` creation into `Get` and then pass the already parsed object into OpGet. You do not need to create it so far in the flow.